pkg/blackboxexporter/blackboxexporter.go

```diff
@@ -0,0 +1,243 @@
+package blackboxexporter
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	"github.com/openshift/route-monitor-operator/api/v1alpha1"
+	"github.com/openshift/route-monitor-operator/pkg/const/blackbox"
+	"github.com/openshift/route-monitor-operator/pkg/util/reconcile"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	//api's used
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	//local packages
+)
+
+type BlackboxExporter struct {
+	Client client.Client
+	Log    logr.Logger
+	Ctx    context.Context
+}
+
+func NewBlackboxExporter(client client.Client, log logr.Logger, ctx context.Context) *BlackboxExporter {
+	return &BlackboxExporter{client, log, ctx}
```

Comment:
didn't know you could do that... does the order of these matter?

Reply:
Yes